examples/runc-hook/main.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+	ocispec "github.com/opencontainers/runtime-spec/specs-go"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/runcfanotify"
+)
+
+var (
+	verbose      = flag.Bool("verbose", false, "verbose")
+	hookPreStart = flag.Bool("hook-prestart", false, "hook PreStart")
+	hookPostStop = flag.Bool("hook-poststop", false, "hook PostStop")
+	cmd          = flag.String("cmd", "", "the command")
+	env          = flag.String("env", "", "the environ")
+	dir          = flag.String("dir", "", "dir")
+	timeout      = flag.String("timeout", "10s", "timeout")
+)
+
+func callback(notif runcfanotify.ContainerEvent) {
+	if *verbose {
+		if notif.Type == runcfanotify.EVENT_TYPE_ADD_CONTAINER {
+			fmt.Printf("Container added: %v pid %d\n", notif.ContainerID, notif.ContainerPID)
+			if notif.ContainerConfig != nil {
+				b, err := json.MarshalIndent(notif.ContainerConfig, "", "  ")
+				if err != nil {
+					fmt.Printf("%s\n", err)
+				}
+				fmt.Printf("%s\n", string(b))
```

Comment:
As we talked, probably have a different param to choose if printing the config.json or not.